backend/layers/business/business.py

```diff
@@ -0,0 +1,186 @@
+
+
+@dataclass
+class CollectionQueryFilter:
+    visibility: Optional[str] # Use an enum
+    owner: Optional[str]
+    # TODO: add list of fields to be returned (if needed)
+
+@dataclass
+class UserInfo:
+    user_id: str
+    token: str
+
+
+class BusinessLogicInterface:
+
+    # Get_collections
+    # Replaces get_collections_list and get_collections_index
+    # Accepts a CollectionFilter class (or kwargs) with:
+    # Date bounds
+    # Visibility
+    # Ownership
+    # List of fields to be returned
+    # Accepts an UserInfo object
+    # Returns a list of dictionaries that only includes the selected fields for each collection
+    # It should NOT add any information that is required by the current API for compatibility reasons (e.g. access_write). These will be delegated to the upper layer
+    # It should NOT do any operation that is required by Curation API assumptions (e.g. remove None values)
+
+    def get_collections(self, filter: CollectionQueryFilter, user_info: UserInfo) -> Iterable[CollectionVersion]:
+        pass
+
+
+    # Get_collection
+    # Replaces get_collection_details
+    # Returns a single collection, with no filtering options
+    # Accepts:
+    # Collection_id
+    # UserInfo for validation
+    # Should reuse most of of the code from the method above
+
+    def get_collection(self, collection_id: str, user_info: UserInfo) -> CollectionVersion:
+        pass
+
+    def get_collection_version(self, version_id: str, user_info: UserInfo) -> CollectionVersion
+        pass
+
+    # Create_collection
+    # Replaces the current create_collection
+    # Accepts:
+    # A dictionary (or Class) with the body of the collection to be created
+    # UserInfo
+    # Should validate the body accepted as param (see existing verify_collection_body)
+    # Should call CrossrefProvider to retrieve publisher metadata information
+    # This method currently collects errors in a list, which will be piped upstream to the API response. This is a good idea but it should be refactor into a generalized pattern (otherwise we’ll “pollute” the business layer with logic specific to the API layer).
+
+    def create_collection(self, collection_metadata: CollectionMetadata, user_info: UserInfo) -> CollectionVersion:
+        pass
+
+    # Delete_collection
+    # Replaces the current delete_collection
+    # Accepts:
+    # Collection_id
+    # UserInfo
+    # Performs authorization on user/collection
+
+    def delete_collection(self, collection_id: str, user_info: UserInfo) -> None:
+        pass
+
+    # Update_collection
+    # Replaces the current update_collection
+    # Accepts:
+    # Collection_id
+    # UserInfo
+    # A dataclass with the body to be updated
+    # Should validate the body
+    # Should handle DOI updates (re-use the existing logic with minimal refactors)
+    # Can either return nothing or the metadata of the updated collection
+
+    # TODO: body should be a dataclass?
```

Comment:
can't it be `CollectionMetadata`? If not using that because it also includes `links` then maybe `links` should be treated as `publisher_metadata`, i.e. moved up a level into CollectionVersion and updated via a separate method.

Reply:
Unfortunately no, `owner` cannot be changed via an update.